landoapi/uplift.py

```diff
@@ -143,9 +160,25 @@ def get_uplift_conduit_state(
             f"Every uplifted patch must have an associated bug ID: {missing} do not."
         )
 
+    rev_ids = [revision["id"] for revision in stack_data.revisions.values()]
+    rev_ids_to_all_diffs = get_rev_ids_to_diffs(phab, rev_ids)
+
     stack = RevisionStack(set(stack_data.revisions.keys()), edges)
 
-    return stack_data, stack, target_repo
+    return stack_data, stack, target_repo, rev_ids_to_all_diffs
+
+
+def get_latest_good_binary_diff(diffs: list[dict]) -> dict:
+    """Given a list of diff dicts, return the latest diff that avoids bug 1865760."""
+    # Iterate through the diffs in order of the latest IDs.
+    for diff in sorted(diffs, key=itemgetter("id"), reverse=True):
+        # Diffs with a `creationMethod` of `commit` may have bad binary data.
+        if diff["creationMethod"] == "commit":
+            continue
+
+        return diff
```

Comment:
nit: I think it would be a little easier to follow what is happening here if we first filter out unwanted records, then return the first/last desired one left, outside the for loop.
```suggestion
    diffs = sorted(diffs, key=itemgetter("id"))
    diffs = [diff for diff in diffs if diff["creationMethod"] != commit]
    if diffs:
        return diffs[-1]
```